### PR TITLE
adding tests to package + some fixes

### PR DESCRIPTION
**Pull request by MonssafToukal**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+using ADNLPModels
+using OptimizationProblems, OptimizationProblems.ADNLPProblems
+using JSOSolvers
 using Test
 
-@testset "BBModels.jl" begin
-  # Write your tests here.
+T = Float64
+n = 5
+problems = (eval(p)(type=Val(T)) for (_,p) ∈ zip(1:n,filter(x -> x != :ADNLPProblems && x != :scosine, names(OptimizationProblems.ADNLPProblems))))
```
[the remark this answers is not in the document]
Comment (tmigot) [reply test/runtests.jl:11]: What is wrong with `scosine`?

File: test/bbmodels_test.jl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+function aux_func(p_metric::ProblemMetrics)
+  median_time = median(get_times(p_metric))
+  memory = get_memory(p_metric)
+  solved = get_solved(p_metric)
+  counters = get_counters(p_metric)
+  return median_time + memory + counters.neval_obj + (Float64(!solved) * 5.0 * median_time)
+end
+
+@testset "Testing BBModels core functions" verbose=true begin
```
Comment (tmigot) [inline test/bbmodels_test.jl:13]: Can we call `obj(nlp, nlp.meta.x0)`?

**Review by tmigot — CHANGES_REQUESTED** [review]
It looks really good. I made a couple of comments to improve it.

File: test/bbmodels_test.jl [hunk 3]
```diff
@@ -0,0 +1,26 @@
+function solver_func(nlp::AbstractNLPModel, p::NamedTuple)
+  return lbfgs(nlp; verbose=0, max_time=60.0)
```
[the remark this answers is not in the document]
Comment (MonssafToukal) [reply test/bbmodels_test.jl:2]: `lbfgs` doesn't take a Named Tuple normally. It was a modification I made in `JSOSolvers.jl` to make it work. I think we can't fully test this functionality until we have parameters working on the solver side. I can always add my version of JSOSolvers and use it to test the parameters are being passed correctly.

[the remark this answers is not in the document]
Comment (tmigot) [reply test/bbmodels_test.jl:2]: I understand. Can't you just add `mem = p[:mem]` when call `lbfgs`?

**Pull request merged**